base/src/uart/uart.cpp

```diff
@@ -0,0 +1,196 @@
+const UartFrameConfig UartController::DEFAULT_FRAME = {
+    .mWordLength = EIGHT_BITS,
+    .mParity = false,
+    .mParityOdd = false,
+    .mStopBits = ONE_STOP_BIT
+};
+
+///
+/// Constructor
+/// @param usart Pointer to usart registers
+/// @param usart_num Usart numbered
+///
+UartController::UartController(uint8_t usart_num):
+mUsartNum(usart_num),
+mBaudRate(UartController::DEFULT_BAUD),
+mFramingConfig(UartController::DEFAULT_FRAME)
+{
+    switch(mUsartNum){
+        case 1:
+            mUsart = USART1;
+            RCC->APB2RSTR |= 1 << 4; // Reset.
+            mBusFreq = APB2_TARGET;
+            RCC->APB2RSTR &= !(1 << 4); // unReset.
+            RCC->APB2ENR |= 1 << 4; // Start.
+            break;
+
+        case 2:
+            mUsart = USART2;
+            RCC->APB1LRSTR |= 1 << 17; // Reset.
+            mBusFreq = APB1_TARGET;
+            RCC->APB2RSTR &= !(1 << 17); // unReset.
+
+            RCC->APB1LENR |= 1 << 17; // Start.
+            break;
+
+        case 3:
+            mUsart = USART3;
+            RCC->APB1LRSTR |= 1 << 18; // Reset.
+            mBusFreq = APB1_TARGET;
+            RCC->APB1LRSTR &= !(1 << 18); // unReset.
+
+            RCC->APB1LENR |= 1 << 18; // Start.
+            break;
+    
+        case 4:
+            mUsart = UART4;
+            RCC->APB1LRSTR |= 1 << 19; // Reset.
+            mBusFreq = APB1_TARGET;
+            RCC->APB1LRSTR &= !(1 << 19); // unReset.
+
+            RCC->APB1LENR |= 1 << 19; // Start.
+            break;
+            
+        case 5:
+            mUsart = UART5;
+            RCC->APB1LRSTR |= 1 << 20; // Reset.'
+            mBusFreq = APB1_TARGET;
+            RCC->APB1LRSTR &= !(1 << 20); // unReset.
+
+            RCC->APB1LENR |= 1 << 20; // Start.
+            break;
+
+        case 6:
+            mUsart = USART6;
+            RCC->APB2RSTR |= 1 << 5; // Reset.
+            mBusFreq = APB2_TARGET;
+            RCC->APB2RSTR &= !(1 << 5); // unReset.
+
+            RCC->APB2ENR |= 1 << 5; // Start.
+            break;
+
+        case 7:
+            mUsart = UART7;
+            RCC->APB1LRSTR |= 1 << 30; // Reset.
+            mBusFreq = APB1_TARGET;
+            RCC->APB1LRSTR &= !(1 << 30); // unReset.
+
+            RCC->APB1LENR |= 1 << 30; // Start.
+            break;
+            
+        case 8:
+            mUsart = UART8;
+            RCC->APB1LRSTR |= 1 << 31; // Reset.
+            RCC->APB1LRSTR &= !(1 << 31); // unReset.
+
+            RCC->APB1LENR |= 1 << 31; // Start.
+            break;
+
+
+    }
+
+    mUsart->CR1 =  // M bits control word length
+        1 << 29 |  // Enable fifo.
+        0 << 28 |  // set M1
+        0 << 12 |  // set MO
+        1 << 2;    // Receiver enable
+
+    setBaudRate(mBaudRate); // Configure baud rate
+    setFraming(mFramingConfig); // Configure framing
+
+}
+
+
+///
+/// Set the baud rate
+/// @param baud_rate_set Baud rate to set to.
+/// @todo baud rate is doubled? Should fix that...
```

Comment:
I think this comment can be removed now that the doubled baud rate issue is resolved

Reply:
Modified.